lib/ansible/module_utils/ordereddict.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (c), Matt Martz <sivel@ansible.com> 2018
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from collections import MutableMapping
+
+from ansible.module_utils.six.moves import map as _imap
+from operator import eq as _eq
+
+try:
+    from collections import KeysView, ValuesView, ItemsView
+except ImportError:
+    from collections.abc import KeysView, ValuesView, ItemsView
+
+try:
+    from ansible.module_utils.six.moves._thread import get_ident as _get_ident
+except ImportError:
+    from ansible.module_utils.six.moves._dummy_thread import get_ident as _get_ident
+
+
+class OrderedDict(dict):
+    'Dictionary that remembers insertion order'
+    # An inherited dict maps keys to values.
+    # The inherited dict provides __getitem__, __len__, __contains__, and get.
+    # The remaining methods are order-aware.
+    # Big-O running times for all methods are the same as regular dictionaries.
+
+    # The internal self.__map dict maps keys to links in a doubly linked list.
+    # The circular doubly linked list starts and ends with a sentinel element.
+    # The sentinel element never gets deleted (this simplifies the algorithm).
+    # Each link is stored as a list of length three:  [PREV, NEXT, KEY].
+
+    def __init__(*args, **kwds):
+        '''Initialize an ordered dictionary.  The signature is the same as
+        regular dictionaries, but keyword arguments are not recommended because
+        their insertion order is arbitrary.
+
+        '''
+        if not args:
+            raise TypeError("descriptor '__init__' of 'OrderedDict' object "
+                            "needs an argument")
+        self = args[0]
+        args = args[1:]
```

Comment:
To solve the no-method-argument pylint warning, change the definition to
```
def __init__(self, *args, **kwds):
```
and remove lines 38-41 (the lines that sanity check and set self from args)

Reply:
*remove lines 38-42.